app/upgrades.go

```diff
@@ -8,22 +8,128 @@ import (
+		"cro1sjcrmp0ngft2n2r3r4gcva4llfj8vjdnefdg4m", // ecosystem
+	},
+	"testnet-croeseid-4": {
+		"tcro14thaw89nlpp8hjm83z6zp3w5ymlpgg2zejncw7",
+		"tcro19uhea66tnx78r5258sq5vdad8msk47w6vn8f06",
+	},
+	"crypto-org-chain-mainnet-dryrun-1": {
+		"cro1h704kvqdh48jzge7vvxpej9d6r9usvssehmxac",
+		"cro1gv6e77tq7l06904g9nuu4nvnwcynaannwjpuaj",
+		"cro160rhmah7kmfy9vg9jklkdqyv6nu9j7jnjpun9j",
+	},
+	"crypto-org-chain-mainnet-1": {
+		"cro160rhmah7kmfy9vg9jklkdqyv6nu9j7jnjpun9j",
+	},
+}
+
 func (app *ChainApp) RegisterUpgradeHandlers(cdc codec.BinaryCodec) {
-	planName := "v5.0"
+	planName := "v6.0.0"
+
+	// Set param key table for params module migration
+	for _, subspace := range app.ParamsKeeper.GetSubspaces() {
+		var keyTable paramstypes.KeyTable
+		switch subspace.Name() {
+		case authtypes.ModuleName:
+			keyTable = authtypes.ParamKeyTable() //nolint:staticcheck
+		case banktypes.ModuleName:
+			keyTable = banktypes.ParamKeyTable() //nolint:staticcheck
+		case stakingtypes.ModuleName:
+			keyTable = stakingtypes.ParamKeyTable() //nolint:staticcheck
+		case minttypes.ModuleName:
+			keyTable = minttypes.ParamKeyTable() //nolint:staticcheck
+		case distrtypes.ModuleName:
+			keyTable = distrtypes.ParamKeyTable() //nolint:staticcheck
+		case slashingtypes.ModuleName:
+			keyTable = slashingtypes.ParamKeyTable() //nolint:staticcheck
+		case govtypes.ModuleName:
+			keyTable = govv1.ParamKeyTable() //nolint:staticcheck
+		case ibcexported.ModuleName:
+			keyTable = ibcclienttypes.ParamKeyTable()
+			keyTable.RegisterParamSet(&ibcconnectiontypes.Params{})
+		case ibctransfertypes.ModuleName:
+			keyTable = ibctransfertypes.ParamKeyTable()
+		case icacontrollertypes.SubModuleName:
+			keyTable = icacontrollertypes.ParamKeyTable()
+		case icahosttypes.SubModuleName:
+			keyTable = icahosttypes.ParamKeyTable()
+		default:
+			continue
+		}
+		if !subspace.HasKeyTable() {
+			subspace.WithKeyTable(keyTable)
+		}
+	}
+	baseAppLegacySS := app.ParamsKeeper.Subspace(baseapp.Paramspace).WithKeyTable(paramstypes.ConsensusParamsKeyTable())
+
 	app.UpgradeKeeper.SetUpgradeHandler(planName, func(ctx context.Context, plan upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
+		sdkCtx := sdk.UnwrapSDKContext(ctx)
+
+		// OPTIONAL: prune expired tendermint consensus states to save storage space
+		if _, err := ibctmmigrations.PruneExpiredConsensusStates(sdkCtx, cdc, app.IBCKeeper.ClientKeeper); err != nil {
+			return nil, err
+		}
+
+		// explicitly update the IBC 02-client params, adding the localhost client type
+		var allowedClients []string
+		ibcParamSpace := app.GetSubspace(ibcexported.ModuleName)
+		ibcParamSpace.Get(sdkCtx, ibcclienttypes.KeyAllowedClients, &allowedClients)
+		allowedClients = append(allowedClients, ibcexported.Localhost)
+		ibcParamSpace.SetParamSet(sdkCtx, &ibcclienttypes.Params{
+			AllowedClients: allowedClients,
+		})
+		if err := ibcmigrationsv6.MigrateICS27ChannelCapability(
```

Comment:
Do you have any context for this line?

Reply:
It can be found here
https://github.com/cosmos/ibc-go/blob/v7.0.0/docs/migrations/v5-to-v6.md#upgrade-proposal